src/main/java/org/opensearch/commons/authuser/User.java

```diff
@@ -250,4 +253,9 @@ public List<String> getCustomAttNames() {
     public String getRequestedTenant() {
         return requestedTenant;
     }
+
+    public static boolean isSuperUser(User user, Settings settings) {
+        List<String> adminDns = settings.getAsList(ConfigConstants.OPENSEARCH_SECURITY_AUTHCZ_ADMIN_DN, Collections.emptyList());
+        return adminDns.contains(user.getName());
```

Comment:
I think we might need some backward compatibility? What happens if user is null? then it will throw exception in line 259. And I'm not a security person, so I don't know how to handle this situation. Just wanted to call out here.

Reply:
Added null check

Reply:
Usually when the user is super admin, the user object is null. [Reference](https://github.com/opensearch-project/alerting/blob/main/alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportGetAlertsAction.kt#L227)

Reply:
Please see details [here](https://github.com/opensearch-project/security/issues/3413#issuecomment-1749928762). 

There are 1 cases of superadmin:

1. Plugin stashed the threadcontext (user == null)
2. User connects to cluster using admin certificate <- scope of this PR

Reply:
Capturing discussion on this PR:

1. User is null means 1 of 2 things - 1) Security is not enabled or 2) plugin is operating in a stashContext block
2. User is non-null then - 1) User is a regular user (including anonymous) or 2) User is a superuser (meaning request comes from admin cert)

This PR pertains to 2) above and also relies on:

1. When security is enabled a user will always be present in the threadcontext
2. The plugin dev will ensure they are not making any calls to read the currently authenticated user from within a stashContext block